wave_benchmark/src/trajectory_compare.cpp

```diff
@@ -0,0 +1,44 @@
+#include "wave/benchmark/trajectory_compare.hpp"
+
+namespace wave {
+
+void TrajectoryCompare::reset() {
+    this->ground_truth.clear();
+    this->measurements.clear();
+    this->error.clear();
+}
+
+void TrajectoryCompare::push_measurement(Pose pose, const TimeType time) {
+    measurements.insert(PoseMeasurement(time, this->measurement_key, pose));
```

Comment:
If I'm not (too lazy to double check), think of emplace as a short hand direct access to the objects constructor, the `measurements` already knows its storing `PoseMeasurements` therefore using emplace you don't have to write

    measurements.insert(PoseMeasurement(time, ....

emplace allows you to simplify it to 

        measurements.emplace(time, ....